src/lib/utils.py

```diff
@@ -77,6 +77,7 @@ def grouped_transform(
     transform: Callable,
     skip: List[str] = None,
     prefix: Tuple[str, str] = None,
+    compat: bool = True,
```

Comment:
It would help to document the parameters.  Perhaps some of these might be obvious to someone who is familiar with the context, but something like "compat" might still be confusing.  Is this short for "compatible"?  Compatible with what?

Reply:
This `compat` argument is only temporary, once this PR goes in I plan on refactoring all usages of `grouped_diff` and `grouped_cumsum` so it will no longer be needed. I might slightly change the behavior of this function then and potentially mark it as private since it should no longer be needed outside of the `utils` module. For that reason, I'm holding off on documenting this better :-)